### PR TITLE
Fixing bugs

### PR DESCRIPTION
**Pull request by Rayden440**
- Refactored electron's "check for update" and "download" code to make it a lot simpler for React.
- Added play button lock to prevent user from spamming it and firing update function multiple times.
- Disabled custom title bar for now until fix at a later date
- Changed window size to get rid of scroll bar

### CODE REVIEW

File: electron/launcher/src/_reducers/patcher.reducer.js [hunk 1]
```diff
@@ -11,17 +11,28 @@ const INITIAL_STATE = {
...
+      return { ...state, playButtonLock: true}
+    case types.INITIAL_CHECK_COMPLETE:
+      return { ...state, reqInitialCheck: false}
+    case types.QUEUE_GAME_LAUNCH:
+      return { ...state, gameLaunchQueued: true}
     case types.IS_LATEST_VERSION:
-      return { ...state, isLatest: true };
-    case types.UPDATE_AVAILABLE:
-      return { ...state, isLatest: false };
+      return { ...state, playButtonLock: false, patching: false};
```
Comment (BenjixD) [inline electron/launcher/src/_reducers/patcher.reducer.js]: you are guaranteed that if you arent patching and play button lock is false, then latest version is true?

File: electron/launcher/src/components/Header/Header.js [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
-      }
+  useEffect(() => {
+    if (props.patch.gameLaunchQueued){
+      props.resetGameLaunchQueue();
+      props.startGameClient();
     }
+  }, [props.patch.gameLaunchQueued]);
+
+  const handleButtonClick = () => {
+    props.checkForUpdate({
```
Comment (BenjixD) [inline electron/launcher/src/components/Header/Header.js]: do we still disable button during patching?

File: electron/launcher/src/components/TitleBar.js [hunk 3]
```diff
@@ -1,30 +1,39 @@
-import React from 'react';
+import React, { useEffect ,useState} from 'react';
 import closeIcon from 'assets/icons/close_icon.svg';
 import minimizeIcon from 'assets/icons/minimize_icon.svg';
 
-const { remote } = window.require('electron')
+const { electron } = window.require('electron')
 
 const TitleBar = props => {
```
Comment (BenjixD) [inline electron/launcher/src/components/TitleBar.js]: do we need to comment out TitleBar if you already commented out the place it was used?

Comment (allenaerostar) [reply electron/launcher/src/components/TitleBar.js]: I think hes saving it for later to add it in.

File: electron/launcher/src/_reducers/patcher.reducer.js (hunk 1 above)
Comment (Rayden440) [reply electron/launcher/src/_reducers/patcher.reducer.js]: No, if someone leaves the launcher on for a long time they may eventually become out of date. However, we always check for update before launching the game. In `Header.js` where we launch the game, you can see we launch the game by checking for update but telling the update function to queue an action (launch game).

File: electron/launcher/src/components/Header/Header.js (hunk 2 above)
Comment (Rayden440) [reply electron/launcher/src/components/Header/Header.js]: Button remains disabled the entire time we check for update, including the time patch is being downloaded.

**Pull request closed without merging**